cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1734,7 +1812,7 @@ class ObjectOutputStream final : public io::OutputStream {
         return Status::OK();
       }
 
-      // Upload current buffer
+      // Upload current buffer if we are above threshold for multi-part upload
```

Comment:
The comment is misleading: this always uploads the current buffer, right?

Reply:
I mean, `CommitCurrentPart` starts by calling `CreateMultipartUpload`.

Reply:
You're right, I just wanted to make the point that we won't call this when we haven't accumulated enough data. I've clarified the comment. [7748824](https://github.com/apache/arrow/pull/41564/commits/77488243b71be6688e9131efac03e9b98d14b1bd)